docs/en/sql-reference/statements/select/with.md

```diff
@@ -4,13 +4,13 @@ toc_title: WITH
 
 # WITH Clause {#with-clause}
 
-This section provides support for Common Table Expressions ([CTE](https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL)), so the results of `WITH` clause can be used in the rest of `SELECT` query.
+This section provides support for Common Table Expressions ([CTE](https://en.wikipedia.org/wiki/Hierarchical_and_recursive_queries_in_SQL)), so the results of `WITH` clause can be used in the rest of `SELECT` query. Named subqueries can be included to the current and children query context. Those identifiers can appear in places where table objects are allowed. Recursion is prevented by hiding the current level CTEs from the WITH expression.
```

Comment:
ClickHouse code provides support for some functionality. The ClickHouse documentation describes the functionality.

Do we talk only about named subqueries in WITH clause, or we talk about named entities, which can be expressions or queries? 

Instead of "those identifiers", I'd use something like "You can use the identifiers, defined in WITH clause,  in places where table objects are allowed".